src/mpl2/src/SACoreSoftMacro.cpp

```diff
@@ -327,6 +359,14 @@ void SACoreSoftMacro::calBoundaryPenalty()
     return;
   }
 
+  int tot_num_macros = 0;
+  for (const auto& macro : macros_) {
+    tot_num_macros += macro.getNumMacro();
+  }
```

Comment:
The naming here is confusing.  'macro' sounds like a single macro not a container of macros